pinot-plugins/pinot-file-system/pinot-gcs/src/main/java/org/apache/pinot/plugin/filesystem/GcsPinotFS.java

```diff
@@ -204,7 +205,7 @@ public boolean delete(URI segmentUri, boolean forceDelete) throws IOException {
     LOGGER.info("Deleting uri {} force {}", segmentUri, forceDelete);
     try {
       if (!exists(segmentUri)) {
-        return false;
+        return forceDelete;
```

Comment:
Should we return `true` here?

Reply:
Even if force delete is false? I thought of it as similar to `rm` vs `rm -f`. But I can change it, lmk what you think.